.github/CONTRIBUTING.md

```diff
@@ -17,40 +16,103 @@ We prefer questions and support requests be posted in the [forum](http://forum.t
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+The footer should contain a [closing reference to an issue](https://help.github.com/articles/closing-issues-via-commit-messages/) if any.
+
+Samples: (even more [samples](https://github.com/angular/angular/commits/master))
+
+```
+docs(changelog): update changelog to beta.5
+```
+```
+fix(release): need to depend on latest rxjs and zone.js
+
+The version in our package.json gets copied to the one we publish, and users need the latest of these.
+```
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
+
+* **build**: Changes that affect the build system or external dependencies (example scopes: gradle, git, idea)
+* **ci**: Changes to our CI configuration files and scripts (example scopes: Jenkins)
+* **docs**: Documentation only changes
+* **feat**: A new feature
+* **fix**: A bug fix for the game
+* **perf**: A code change that improves performance of the game
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, copyright, etc)
+* **test**: Adding missing tests or correcting existing tests
+
+### Scope
+The scope should be the name of the package(s) affected (as perceived by the person reading the changelog generated from commit messages).
+The package prefix `org.terasology` should be ommitted.
 
-1. If your work spans multiple local commits make sure they can be merged directly into the target branch. If you have some commit and merge noise in the history have a look at git-rebase. Ideally, every commit should be able to be used in isolation -- that is, each commit must build and pass all tests.
-1. The first line of the commit message should be a descriptive sentence about what the commit is doing. It should be possible to fully understand what the commit does by just reading this single line.
-1. **Optional:** Following the single line description (ideally no more than 70 characters long) should be a blank line followed by a list with the details of the commit.
-1. **Optional:** Add `Fix/Fixing/Fixes/Close/Closing/Refs #issue` if you want to mark the issue mentioned as complete in the issue tracker when the change is merged.
+Samples:
+```
+feat(world): add JOML API to WorldProvider
+```
+
+There are currently a few exceptions to the "use package name" rule:
+
+> :construction: What are our exceptions? Non-code changes (build, ci, ...)?
```

Comment:
I wonder if package would be too rigid, especially if a change affects multiple. Could it be more loosely defined concepts, like "JOML", then we can provide a list of initial common ones to pick from or come up with your own if it is different and unique, yet makes up a category likely to see more activity?

Reply:
Hm, I'm a bit hesitant to use "arbitrary" names here. We already have some confusion whether something is 
```
build: fix the gradle script
```
or 
```
fix(gradle): correct script
```
or
```
chore(gradle): fix the script
```

To my understanding this _scope_ should describe the part of the code base that has changed/is affected, the topic (such as JOML) would be mentioned in the subject line. I'd also organize these topics rather via labels or boards in Github.

Maybe we can limit the scope to the _top-level package_ (or max one level more), so you'd have `feat(world)` or `feat(world/chunks)`, but not something lengthy like `feat(world/chunks/localChunkProvider)` taking up the whole line...